From 6f8623c570247903e3cba925676677e44a99c69c Mon Sep 17 00:00:00 2001
From: Eduardo Otubo <otubo@redhat.com>
Date: Tue, 28 Apr 2020 08:22:05 +0200
Subject: [PATCH 2/3] swap file "size" being used before checked if str (#315)

RH-Author: Eduardo Otubo <otubo@redhat.com>
Message-id: <20200422130428.7663-3-otubo@redhat.com>
Patchwork-id: 96033
O-Subject: [RHEL-7.7.z/RHEL-7.8.z cloud-init PATCH 2/3] swap file "size" being used before checked if str (#315)
Bugzilla: 1801094
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>
RH-Acked-by: Cathy Avery <cavery@redhat.com>

commit 46cf23c28812d3e3ba0c570defd9a05628af5556
Author: Eduardo Otubo <otubo@redhat.com>
Date:   Tue Apr 14 18:16:25 2020 +0200

    swap file "size" being used before checked if str (#315)

    Swap file size variable was being used before checked if it's set to str
    "auto". If set to "auto", it will break with:

      failed to setup swap: unsupported operand type(s) for /: 'str' and 'int'

    Signed-off-by: Eduardo Otubo <otubo@redhat.com>
    RHBZ: 1772505

Signed-off-by: Eduardo Otubo <otubo@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 cloudinit/config/cc_mounts.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cloudinit/config/cc_mounts.py b/cloudinit/config/cc_mounts.py
index 6884ddf..811781f 100644
--- a/cloudinit/config/cc_mounts.py
+++ b/cloudinit/config/cc_mounts.py
@@ -274,7 +274,6 @@ def setup_swapfile(fname, size=None, maxsize=None):
     maxsize: the maximum size
     """
     swap_dir = os.path.dirname(fname)
-    mibsize = str(int(size / (2 ** 20)))
     if str(size).lower() == "auto":
         try:
             memsize = util.read_meminfo()['total']
@@ -286,6 +285,7 @@ def setup_swapfile(fname, size=None, maxsize=None):
         size = suggested_swapsize(fsys=swap_dir, maxsize=maxsize,
                                   memsize=memsize)
 
+    mibsize = str(int(size / (2 ** 20)))
     if not size:
         LOG.debug("Not creating swap: suggested size was 0")
         return
-- 
1.8.3.1

